pylint_django/augmentations/__init__.py

```diff
@@ -644,25 +644,6 @@ def is_model_view_subclass_method_shouldnt_be_function(node):
     return parent is not None and node_is_subclass(parent, *subclass)
 
 
-def is_model_view_subclass_unused_argument(node):
-    """
-    Checks that node is get or post method of the View class and it has valid arguments.
-
-    TODO: Bad checkings, need to be more smart.
-    """
-    if not is_model_view_subclass_method_shouldnt_be_function(node):
-        return False
-
-    return is_argument_named_request(node)
-
-
-def is_argument_named_request(node):
-    """
-    If an unused-argument is named 'request' ignore that!
-    """
-    return 'request' in node.argnames()
```

Comment:
I am wondering if we could somehow check the name of the offending argument instead of everything else. Need to dig deeper into that.

Reply:
So you want to pursue a different implementation instead of extending  Pylints`ignored-argument-names` setting?

Reply:
I'm not sure, this is why I was asking. I can see your changes make sense, however I would prefer a smaller diff (e.g. something along the lines of if 1st argument is named 'request') if at all possible. I need a few days to fix the test suite first and then check this specific area of pylint   before I can form an opinion which solution is more preferable.

Reply:
Okay, no problem. Appending to the regex pattern also feels a bit dirty, in my opinion.

But I took a cursory look into a different solution and, as far as I can tell, I don't think it is possible to suppress the warning for a specific argument using the existing functions:

- The `unused-argument` warning is generated [in `_check_unused_arguments`](https://github.com/PyCQA/pylint/blob/97f4f2ae187df933f072d74fd8347ec14213f5de/pylint/checkers/variables.py#L1616), which is called by `_check_is_unused`, which in turn is called by `VariablesChecker.leave_functiondef`.
  - So unfortunately there is no `VariablesChecker.leave_argumentdef` or something we could hook into and suppress the warning for specific arguments. The suppression has to happen for function definitions.
- `suppress_message` could be another possible point for suppressing the warning for specific arguments
   - Unfortunately there is no way to pass additional information to the [internal suppression logic](https://github.com/PyCQA/pylint-plugin-utils/blob/db903068d4ad64d919c014af097cdafadcf1748d/pylint_plugin_utils/__init__.py#L82). It only uses the name of the warning.
      - But the `to_append_args` used in the internal logic contains the name of the argument of `unused-argument` warnings.
   - Extending the `suppress_message` of `pylint-plugin-utils` to allow suppression of warnings based on additional information could be cleaner way to implement this.

Maybe I missed something and I would be happy to be proven wrong :D